src/test/java/io/jenkins/plugins/analysis/core/quality/QualityGateTest.java

```diff
@@ -282,6 +285,261 @@ void shouldBeFailedWhenTotalFailedAndNewUnstableThresholdIsLowerThanWarningCount
+        when(newFailedThresholdResult.isHighReached()).thenReturn(true);
+        when(newFailedThresholdResult.isNormalReached()).thenReturn(true);
+        when(newFailedThresholdResult.isLowReached()).thenReturn(true);
+
+        AnalysisResult analysisResult = mock(AnalysisResult.class);
+        when(analysisResult.getTotalSize()).thenReturn(10);
+
+        testEvaluation(totalThresholdResult, totalUnstableThresholdResult, newFailedThresholdResult,
+                newUnstableThresholdResult, analysisResult, false);
+    }
+
+    @Test
+    void shouldReturnEvaluationForNewUnstable() {
+        ThresholdResult totalThresholdResult = mock(ThresholdResult.class);
+        ThresholdResult totalUnstableThresholdResult = mock(ThresholdResult.class);
+        ThresholdResult newFailedThresholdResult = mock(ThresholdResult.class);
+        ThresholdResult newUnstableThresholdResult = mock(ThresholdResult.class);
+
+        when(newUnstableThresholdResult.isTotalReached()).thenReturn(true);
+        when(newUnstableThresholdResult.isHighReached()).thenReturn(true);
+        when(newUnstableThresholdResult.isNormalReached()).thenReturn(true);
+        when(newUnstableThresholdResult.isLowReached()).thenReturn(true);
+
+        AnalysisResult analysisResult = mock(AnalysisResult.class);
+        when(analysisResult.getTotalSize()).thenReturn(10);
+
+        testEvaluation(totalThresholdResult, totalUnstableThresholdResult, newFailedThresholdResult,
+                newUnstableThresholdResult, analysisResult, true);
+    }
+
+    /**
+     * Test the evaluation if all messages are failure messages.
+     *
+     * @param messages
+     *         collection of evaluation messages
+     */
+    private void testEvaluationMessageAllFailure(Collection<String> messages) {
+        ArrayList<String> evaluations = (ArrayList<String>) messages;
+        for (String message : evaluations) {
+            assertThat(message).contains("FAILURE");
+        }
+    }
+
+    /**
+     * Test the evaluation if all messages are unstable messages.
+     *
+     * @param messages
+     *         collection of evaluation messages
+     */
+    private void testEvaluationMessageAllUnstable(Collection<String> messages) {
+        ArrayList<String> evaluations = (ArrayList<String>) messages;
+        for (String message : evaluations) {
+            assertThat(message).contains("UNSTABLE");
+        }
+    }
+
+    /**
+     * Test the test setup defined by the parameter.
+     *
+     * @param totalThresholdResult
+     *         the threshold result for total
+     * @param totalUnstableThresholdResult
+     *         the threshold result for total unstable
+     * @param newFailedThresholdResult
+     *         the threshold result for new failed
+     * @param newUnstableThresholdResult
+     *         the threshold result for new unstable
+     * @param analysisResult
+     *         the analysis result
+     * @param flag
```

Comment:
Hier wäre ein Enum leichter zu verstehen beim Aufrufer: z.B. `ASSERT_FAILURE`, `ASSERT_UNSTABLE`. Bzw. gleich nur `FAILURE` bzw. `UNSTABLE`, dann kann `name()` benutzt werden, um die Methoden testEvaluationMessageAllUnstable/Failure durch eine mit Parameter zu ersetzen.

Reply:
added enum State and use this to identify which type of message should be checked